In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import confusion_matrix

df = pd.read_csv("C:/Users/Gabo0/Desktop/Jupyter Python/Ejercicios Linkedin/base_datos_2008.csv")

In [2]:
df = df.dropna(subset=["ArrDelay"])
df = df.sample(frac=1).head(100000)
Y = df["ArrDelay"] < 30 # True = 1 / False = 0 // Intepretación: Vuelos sin retraso notable
X = df[["DepDelay"]]

In [3]:
logreg=LogisticRegression()
logreg.fit(X,Y)
Y_pred = logreg.predict(X)

In [4]:
np.round(logreg.predict_proba(X),3)

array([[0.03 , 0.97 ],
       [0.006, 0.994],
       [0.054, 0.946],
       ...,
       [0.018, 0.982],
       [0.085, 0.915],
       [0.008, 0.992]])

In [5]:
np.mean(Y_pred == Y)

0.95601

In [6]:
np.mean(Y)

0.86439

La confusión depende de las x reales y los valores predichos

In [7]:
confusion_matrix = confusion_matrix(Y,Y_pred)
print(confusion_matrix)

[[10204  3357]
 [ 1042 85397]]


LAS FILAS SON LOS VALORES REALES Y LAS COLUMNAS LOS VALORES PREDICHOS

LOS 10204 SON LOS CASOS QUE ERAN CEROS Y EL MODELO LO PREDIJO, LOS 3357 SON LOS CASOS QUE EL VUELO NO TUVO RETRASO Y QUE NUESTRO MODELO DIJO QUE SI.

LOS 85376 SON LOS CASOS QUE LOS VUELOS NO SE RETRASARON Y NUESTRO MODELO HABÍA PREDICHO QUE NO SE HABÍAN RETRASADO.